### PR TITLE
fix(ee): GetKeys should return an error

### PR DESCRIPTION
**Pull request by ajeetdsouza**
Currently, `GetKeys` exits immediately upon error. While this works for most uses, it ends up shutting down the Alpha when a restore request is made with bad input. This PR fixes that.

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7713)

### CODE REVIEW

File: dgraph/cmd/decrypt/decrypt.go [hunk 1]
```diff
@@ -57,17 +56,23 @@ func init() {
 	ee.RegisterEncFlag(flag)
 }
 func run() {
-	opts := options{
-		file:   Decrypt.Conf.GetString("file"),
-		output: Decrypt.Conf.GetString("out"),
+	keys, err := ee.GetKeys(Decrypt.Conf)
+	if err != nil {
```
Comment (jarifibrahim) [inline dgraph/cmd/decrypt/decrypt.go:60]: Use x.Check everywhere.

File: dgraph/cmd/live/run.go [hunk 2]
```diff
@@ -697,8 +697,11 @@ func run() error {
 	}
 
 	creds := z.NewSuperFlag(Live.Conf.GetString("creds")).MergeAndCheckDefault(x.DefaultCreds)
+	keys, err := ee.GetKeys(Live.Conf)
+	if err != nil {
```
Comment (jarifibrahim) [inline dgraph/cmd/live/run.go:701]: use x.Check

File: ee/keys_ee.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	"github.com/dgraph-io/ristretto/z"
+	"github.com/spf13/viper"
+)
+
+// GetKeys returns the ACL and encryption keys as configured by the user
+// through the --acl, --encryption, and --vault flags. On OSS builds,
+// this function always returns an error.
+func GetKeys(config *viper.Viper) (*Keys, error) {
+	keys := &Keys{}
+	var err error
```
Comment (jarifibrahim) [inline ee/keys_ee.go:28]: can this be removed?

**Review by jarifibrahim — APPROVED** [review]
Some minor comments.

File: ee/keys_ee.go (hunk 3 above)
Comment (ajeetdsouza) [reply ee/keys_ee.go:28]: The first use of `err` assigns directly to a struct:

```go
keys.AclKey, err = ioutil.ReadFile(aclKeyFile)
```

So we can't remove the error declaration, unfortunately.

**Pull request merged**